### ISSUE TITLE
Skip cache save feature

### ISSUE THREAD

**Issue by kian99** [actions/setup-go#497]
**Description:**
I'd like to propose a new input to the action to enable skipping cache saves but maintaining the ability to restore an existing cache.
Something like `skip-cache-save`.

**Justification:**
We have the case in our CI/CD where we have the following jobs running in parallel:
1. Run tests - i.e. runs `go test ./...`
2. Run linting - i.e. run `golangci-lint`

Both jobs benefit from Go package caching but in the above scenario, the linting job finishes first and creates a cache that doesn't contain Go test artefacts. It would be useful to only have 1 job create the cache i.e. the job that populates the package cache and the test/build cache.

I imagine this can be handled by using actions/cache, but this addition feels like a simple way of achieving better cache control without dropping to a lower level tool.

**Are you willing to submit a PR?**
Possibly, if the feature request is relevant.

**Comment by priya-kinthali** [comment]
Hello @kian99 👋,
Thank you for this feature request. We will investigate it and get back to you as soon as we have some feedback.

**Comment by adamdecaf** [comment]
Similar to https://github.com/actions/setup-go/issues/495 the caching process takes a long time on our runners. There's no reason to cache by default as Github Actions already has tons of network proxying. 

It takes 8s to setup Go in one job, but 2m45s to cache after the job.. Can we disable this post-job caching by default?

**Comment by washanhanzi** [comment]
In `v5`, use `cache: false`.

```yaml
- uses: actions/setup-go@v5
  with:
    go-version: 'stable'
    cache: false
```

**Comment by gavinsharp** [comment]
This looks like a duplicate of https://github.com/actions/setup-go/issues/316

**Comment by aparnajyothi-y** [comment]
Hello Everyone, The issue reported here has been addressed in #696, which has now been merged into the main branch. We will proceed to close this issue. :)
Please feel free to reach us in case of any further clarifications/concerns :)
